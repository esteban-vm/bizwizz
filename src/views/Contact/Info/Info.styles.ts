import clsx from 'clsx'

const styles = {
  wrapper: clsx('mt-8 flex items-center'),
  icon_container: clsx('mr-[1.6rem]'),
  icon: clsx('h-[2.4rem] w-[2.4rem] fill-primary'),
  detail: clsx('text-left'),
  paragraph: clsx('text-[1.6rem] first:text-3xl first:font-semibold'),
}

export default styles
