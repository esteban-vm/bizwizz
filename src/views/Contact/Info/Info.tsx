import type { FC, ID, IconName } from '@/types'
import { Icon } from '@/components'
import styles from './Info.styles'

export type InfoProps = {
  id: ID
  detail: { subject: string; content: string }
  icon: IconName
}

const Info: FC<InfoProps> = ({ id, detail, icon }) => {
  return (
    <article className={styles.wrapper} id={id}>
      <div className={styles.icon_container}>
        <Icon className={styles.icon} name={icon} />
      </div>
      <div className={styles.detail}>
        <p className={styles.paragraph}>{detail.subject}</p>
        <p className={styles.paragraph}>{detail.content}</p>
      </div>
    </article>
  )
}

export default Info
