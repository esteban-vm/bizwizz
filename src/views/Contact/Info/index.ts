export { default, type InfoProps } from './Info'
