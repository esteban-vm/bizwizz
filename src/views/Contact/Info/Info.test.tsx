import { render, cleanup, within, screen } from '@/tests'
import Info from './Info'
import info from '../Contact.data'

describe('Info:', () => {
  let asFragment: () => DocumentFragment
  const [testInfo] = info

  beforeEach(() => {
    void ({ asFragment } = render(<Info {...testInfo} />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const infoElem = screen.getByRole('article')
    expect(infoElem).toBeInTheDocument()
    expect(infoElem).toHaveAttribute('id', testInfo.id)
    expect(within(infoElem).getByRole('img')).toBeInTheDocument()
    expect(within(infoElem).getByText(testInfo.detail.subject)).toBeInTheDocument()
    expect(within(infoElem).getByText(testInfo.detail.content)).toBeInTheDocument()
  })
})
