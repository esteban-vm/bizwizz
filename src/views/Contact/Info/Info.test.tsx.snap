// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Info: > should render correctly 1`] = `
<DocumentFragment>
  <article
    class="mt-8 flex items-center"
    id="1bcfb7c3-6138-445e-81ab-4109756943d0"
  >
    <div
      class="mr-[1.6rem]"
    >
      <svg
        class="h-[2.4rem] w-[2.4rem] fill-primary"
        role="img"
      >
        <use
          xlink:href="#icon-20"
        />
      </svg>
    </div>
    <div
      class="text-left"
    >
      <p
        class="text-[1.6rem] first:text-3xl first:font-semibold"
      >
        Address
      </p>
      <p
        class="text-[1.6rem] first:text-3xl first:font-semibold"
      >
        Mountain View, California
      </p>
    </div>
  </article>
</DocumentFragment>
`;
